sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -542,18 +550,21 @@
    * <p>If the output type is {@link GenericRecord} use {@link #writeCustomTypeToGenericRecords()}
    * instead.
    */
-  public static <UserT, OutputT> TypedWrite<UserT, Void, OutputT> writeCustomType() {
-    return AvroIO.<UserT, OutputT>defaultWriteBuilder().setGenericRecords(false).build();
+  public static <UserT, OutputT> TypedWrite<UserT, Void, OutputT> writeCustomType(
+      Class<OutputT> recordClass) {
```

Comment:
This is a breaking change. The class needs to be propagated for proper factories instanciation.

Reply:
Can you make it into two codepaths? I am just coming to this PR to help out a little so I don't have all the history of your thoughts on this.

Reply:
I guess it was a question if we can avoid breaking changes here, isn't it?